restalchemy/dm/types.py

```diff
@@ -688,6 +688,18 @@ def to_openapi_spec(self, prop_kwargs):
         return spec
 
 
+class UTCDateTimeZ(UTCDateTime):
+
+    def validate(self, value):
+        return isinstance(value, datetime.datetime) and (
+            value.tzinfo == datetime.timezone.utc
+        )
+
+    def from_simple_type(self, value):
+        result = super(UTCDateTimeZ, self).from_simple_type(value)
+        return result.replace(tzinfo=datetime.timezone.utc)
```

Comment:
Looks like `replace` doesn't convert, it ultimatively changes timezone without time convertation

```
In [11]: u = datetime.datetime.now(ZoneInfo("America/Los_Angeles"))

In [12]: u
Out[12]: datetime.datetime(2025, 3, 4, 0, 28, 17, 534284, tzinfo=zoneinfo.ZoneInfo(key='America/Los_Angeles'))

In [13]: u.replace(tzinfo=datetime.timezone.utc)
Out[13]: datetime.datetime(2025, 3, 4, 0, 28, 17, 534284, tzinfo=datetime.timezone.utc)

In [14]: u
Out[14]: datetime.datetime(2025, 3, 4, 0, 28, 17, 534284, tzinfo=zoneinfo.ZoneInfo(key='America/Los_Angeles'))

In [15]: u.astimezone(datetime.timezone.utc)
Out[15]: datetime.datetime(2025, 3, 4, 8, 28, 17, 534284, tzinfo=datetime.timezone.utc)
```